pointcloud/kdtree.go

```diff
@@ -223,6 +276,82 @@ func (kd *KDTree) KNearestNeighbors(p r3.Vector, k int, includeSelf bool) []*Poi
 	return keeperToArray(keep.Heap, kd.points, p, includeSelf, k)
 }
 
+// KNearestNeighborsConcurrent returns the k nearest points ordered by distance. if includeSelf is true and if the point p
+// is in the point cloud, point p will also be returned in the slice as the first element with distance 0. With concurrency
+func (kd *KDTree) KNearestNeighborsConcurrent(p r3.Vector, k int, includeSelf bool) []*PointAndData {
```

Comment:
I tried to do KNearestNeighbors concurrently but I realized by passing in part of the heap to each thread this would happen. Supposing that I'm looking for 6 neighbors, if I made 4 threads, I would just be left with 24 neighbors. I could then take this list and then find the minimum distance neighbors, but not sure if that would be particularly fruitful.